src/D2L.CodeStyle.TestAnalyzers/ServiceLocator/CustomTestServiceLocatorAnalyzer.cs

```diff
@@ -140,13 +154,37 @@ IMethodSymbol method
 				&& method.Parameters.Length > 0;
 		}
 
-		private bool IsClassWhitelisted( ISymbol classSymbol ) {
+		private bool IsClassWhitelisted(
+			HashSet<string> whitelistedClasses,
+			ISymbol classSymbol
+		) {
 			bool isWhiteListed = _excludeKnownProblems
-				&& CustomServiceLocatorAnalyzerWhitelist
-					.WhitelistedClasses
-					.Contains( classSymbol.ToString() );
+				&& whitelistedClasses.Contains( classSymbol.ToString() );
 
 			return isWhiteListed;
 		}
+
+		private HashSet<string> GetWhitelist(

```

Comment:
nit: you could return `ImmutableHashSet<string>` from here if you use `ImmutableHashSet.CreateBuilder<string>()` instead of `new HashSet<string>()` in the function (and return with `.ToImmutable()`

Reply:
Good thought. I expect I'm going to make some more changes in here anyways, so it seems worth doing.